pymc3/distributions/multivariate.py

```diff
@@ -690,6 +691,144 @@ def logp(self, x):
         )
 
 
+class DirichletMultinomial(Discrete):
+    R"""Dirichlet Multinomial log-likelihood.
+
+    Dirichlet mixture of multinomials distribution, with a marginalized PMF.
+
+    .. math::
+
+    f(x \mid n, \alpha) = \frac{\Gamma(n + 1)\Gamma(\sum\alpha_k)}
+                              {\Gamma(\n + \sum\alpha_k)}
+                         \prod_{k=1}^K
+                         \frac{\Gamma(x_k + \alpha_k)}
+                              {\Gamma(x_k + 1)\Gamma(alpha_k)}
+
+    ==========  ===========================================
+    Support     :math:`x \in \{0, 1, \ldots, n\}` such that
+                :math:`\sum x_i = n`
+    Mean        :math:`n \frac{\alpha_i}{\sum{\alpha_k}}`
+    ==========  ===========================================
+
+    Parameters
+    ----------
+    alpha : two-dimensional array
+        Dirichlet parameter.  Elements must be non-negative.
+        Dimension of each element of the distribution is the length
+        of the second dimension of alpha.
+    n     : one-dimensional array
+        Total counts in each replicate.
+
+    """
+
+    def __init__(self, n, alpha, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if len(self.shape) > 1:
+            self.n = tt.shape_padright(n)
+            self.alpha = tt.as_tensor_variable(alpha) if alpha.ndim > 1 else tt.shape_padleft(alpha)
```

Comment:
@lucianopaz @ricardoV94, not sure if this is quite what you meant, but I do the exact same reshaping as in the Multinomial.

Reply:
No. I meant that in the `Multinomial` Distribution, we used to have a line in `__init__` that did: `n = np.squeeze(n)`. That line caused all sorts of problems when we had `n.ndim > 1`. This situation wasn't covered by existing tests so I wrote up `test_batch_multinomial`. A better approach would be to either parametrize a single test with multiple `n` and `alpha` values that span many shapes, or add test fixtures for `n` and `alpha` to achieve the same result.